### PR TITLE
Adding draft drag & drop API spec

### PR DESCRIPTION
**Pull request by Ram-G**
This API spec is mostly the same as the [UWP drag & drop API](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.datatransfer.dragdrop.core?view=winrt-19041) with a few internal behavior changes. Most notably the change from `DragDropManager::GetForCurrentView` -> `DragDropManager::GetForCurrentWindow` which we are guaranteeing will work not just for CoreWindows/AppWindows as the UWP API does, but also for USER32.DLL HWNDs which means that in theory this API can serve all classic Win32 applications which use HWNDs.

### CODE REVIEW

File: specs/dragdrop-api/dragdrop-api.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+# Remarks
+
+Coming soon.
+
+# API Notes
+
+* `CoreDragDropManager`: Initializes drag & drop environment to enable an application window to be set as a drop target.
+  * `GetForCurrentView`: Static method that constructs a `CoreDragDropManager` for a CoreWindow/AppWindow app.
+  * `GetForHWND`: Static method that constructs a `CoreDragDropManager` for a classic HWND-based windowing app.
```
Comment (Ram-G) [inline specs/dragdrop-api/dragdrop-api.md]: Consider renaming to `GetForWindowId` since MIDL3 I believe doesn't allow HWND as a parameter (currently the parameter to this function is a `UInt32`).

File: specs/dragdrop-api/dragdrop-api.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  * ...
+* `CoreDragOperation`: Object used to create and start a drag operation from your app.
+  * `AllowedOperations`: Use this property to set the complete set of operations the source app wishes to allow in drag and drop scenarios. You can specify zero or more flags. To set a desired default operation, use the `DataPackage.RequestedOperation` property. Users can override this choices by using SHIFT and CTRL keys. In this case, the target app must inspect the key state to determine the operation the user selected.
+  * ...
+* `CoreDragUIContentMode`: Use this to specify whether the `CoreDragOperation` should wait for content load before starting, or run in parallel.
+* `CoreDragUIOverride`
+  * ...
+* `CoreDropOperationTargetRequestedEventArgs`
+  * `SetTarget`: Provide your implementation of `ICoreDropOperationTarget` which specifies the actions your app should take when the user is hovering over your application's window.
+* `ICoreDropOperationTarget`
```
Comment (Ram-G) [inline specs/dragdrop-api/dragdrop-api.md]: Add descriptions here for the contract of `ICoreDropOperationTarget` since it's a very important interface.

File: specs/dragdrop-api/dragdrop-api.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+1. The namespace is now `Microsoft.Reunion.DragDrop`.
+2. The addition of `GetForHWND()` API to register any Windows app as a drop target.
+
+# Examples
+
+## Enabling your application's window to be a drop target
+
+Let's say you are building the next greatest photo organizer app. To add photos to the library, you want to enable users to drag in files, for example from File Explorer to your app. To do this, you need to set your application's HWND/CoreWindow/AppWindow to be a drop target.
+
+1. Get `CoreDragDropManager`:
```
Comment (Ram-G) [inline specs/dragdrop-api/dragdrop-api.md]: Discussion: Consider removing the "Core" naming prefix everywhere in this API.

Comment (usvoyager) [reply specs/dragdrop-api/dragdrop-api.md]: +1 yes

---
In reply to: [459622927](https://github.com/Ram-G/ProjectReunion/pull/1#discussion_r459622927) [](ancestors = 459622927)

File: specs/dragdrop-api/dragdrop.idl [hunk 4]
```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corp. Licensed under the MIT license.
+
+namespace Microsoft.Reunion.DragDrop
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >Microsoft.Reunion.DragDrop [](start = 10, length = 26)

keep it Microsoft.ProjectReunion.ApplicationModel to start with

File: specs/dragdrop-api/dragdrop.idl [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        void SetPointerId(UInt32 pointerId);
+        void SetDragUIContentFromSoftwareBitmap(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap);
+        void SetDragUIContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        Windows.Foundation.IAsyncOperation<Microsoft.Reunion.ApplicationModel.DataPackageOperation> StartAsync();
+    }
+
+    runtimeclass ICoreDragDropManager
+    {
+        CoreDragDropManager GetForCurrentView();
+        CoreDragDropManager GetForHWND(HWND hwnd);
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >CoreDragDropManager GetForHWND(HWND hwnd); [](start = 8, length = 42)

lets remove this, GetForCurrentView() should support all kind of apps, windows and views

File: specs/dragdrop-api/dragdrop.idl [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    enum CoreDragUIContentMode
+    {
+        Auto,
+        Deferred,
+    };
+
+    runtimeclass CoreDragInfo
+    {
+        Microsoft.Reunion.ApplicationModel.DataPackage Data { get; };
+        Microsoft.Reunion.DragDrop.DragDropModifiers Modifiers { get; };
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >Microsoft.Reunion.DragDrop.DragDropModifiers [](start = 8, length = 44)

just DragDropModifiers, it is in the same namespace as CoreDragInfo #Resolved

File: specs/dragdrop-api/dragdrop.idl [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        Windows.Foundation.Point Position { get; };
+        Microsoft.Reunion.ApplicationModel.DataPackage.DataPackageOperation AllowedOperations { get; };
+    }
+
+    runtimeclass CoreDragUIOverride
+    {
+        void SetContentFromSoftwareBitmap(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap);
+        void SetContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        boolean IsContentVisible;
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >boolean [](start = 8, length = 7)

Boolean not boolean #Resolved

Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: hmm, I meant to replace all but missed. thanks

File: specs/dragdrop-api/dragdrop.idl (hunk 5 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: hmm, if we remove this, would we call something like `GetActiveWindow` or `GetForegroundWindow` in the code to identify the correct HWND? Although I understand that implementation concerns are not part of this spec review

File: specs/dragdrop-api/dragdrop.idl [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        void SetDragUIContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        Windows.Foundation.IAsyncOperation<Microsoft.Reunion.ApplicationModel.DataPackageOperation> StartAsync();
+    }
+
+    runtimeclass ICoreDragDropManager
+    {
+        CoreDragDropManager GetForCurrentView();
+        CoreDragDropManager GetForHWND(HWND hwnd);
+        event Windows.Foundation.TypedEventHandler<CoreDragDropManager, CoreDropOperationTargetRequestedEventArgs> TargetRequested;
+        boolean AreConcurrentOperationsEnabled;
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >AreConcurrentOperationsEnabled [](start = 16, length = 30)

do we need to expose it in v1, we should discuss and may be add it in v2

File: specs/dragdrop-api/dragdrop.idl [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    runtimeclass CoreDropOperationTargetRequestedEventArgs
+    {
+        void SetTarget(CoreDropOperationTarget target);
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >SetTarget [](start = 13, length = 9)

Consider: model this as a property instead
CoreDropOperationTarget DropTarget { set; };

File: specs/dragdrop-api/dragdrop.idl (hunk 8 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: good point, I meant to make a comment about this. I'm going to investigate where this is used internally

File: specs/dragdrop-api/dragdrop.idl (hunk 5 above)
Comment (usvoyager) [reply specs/dragdrop-api/dragdrop.idl]: Yes thats the idea, call something like GetActiveWindow or GetForegroundWindow internally, when Reunion WIndow/View type get defined, can provide a overload GetForWindow(WindowRef)

---
In reply to: [459649526](https://github.com/Ram-G/ProjectReunion/pull/1#discussion_r459649526) [](ancestors = 459649526)

File: specs/dragdrop-api/dragdrop.idl [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        CoreDragUIContentMode DragUIContentMode;
+        void SetPointerId(UInt32 pointerId);
+        void SetDragUIContentFromSoftwareBitmap(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap);
+        void SetDragUIContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        Windows.Foundation.IAsyncOperation<Microsoft.Reunion.ApplicationModel.DataPackageOperation> StartAsync();
+    }
+
+    runtimeclass ICoreDragDropManager
+    {
+        CoreDragDropManager GetForCurrentView();
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >GetForCurrentView [](start = 28, length = 17)

GetForCurrentWindow might be appropriate here, View has a confusing meaning

File: specs/dragdrop-api/dragdrop.idl [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+        Microsoft.Reunion.ApplicationModel.DataPackageOperation AllowedOperations;
+        Microsoft.Reunion.ApplicationModel.DataPackage Data { get; };
+        CoreDragUIContentMode DragUIContentMode;
+        void SetPointerId(UInt32 pointerId);
+        void SetDragUIContentFromSoftwareBitmap(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap);
+        void SetDragUIContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        Windows.Foundation.IAsyncOperation<Microsoft.Reunion.ApplicationModel.DataPackageOperation> StartAsync();
+    }
+
+    runtimeclass ICoreDragDropManager
```
Comment (Ram-G) [inline specs/dragdrop-api/dragdrop.idl]: no "I" prefix

File: specs/dragdrop-api/dragdrop-api.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+* XAML/WinUI - Wrapper around UWP drag & drop. [Blog post](https://docs.microsoft.com/en-us/archive/msdn-magazine/2015/august/windows-10-modern-drag-and-drop-for-windows-universal-applications). [Samples](https://github.com/microsoft/Windows-universal-samples/tree/master/Samples/XamlDragAndDrop).
+* WPF - Wrapper around OLE drag & drop. [Documentation](https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/drag-and-drop-overview).
+
+NOTE: It is important to note that use of a UI framework is _not_ necessary to implement drag & drop functionality in an app. An app can implement drag & drop by directly calling one of the two OLE or UWP APIs.
+
+## Motivation for Project Reunion drag & drop
+
+1. **Bring modern drag & drop to all classic apps in an easy-to-incorporate package:** Foremost among the issues we want to address is that the existing drag & drop APIs can be cumbersome to use; especially the older OLE API, and its wrappers such as WPF. In addition, the OLE API is lacking in modern, illustrative UI. With this Project Reunion API, we plan to bring the ability to access modern drag & drop for _all_ apps.
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: > the OLE API is lacking in modern, illustrative UI

What do you mean by this? Do you mean there are no sample apps or example codes?

File: specs/dragdrop-api/dragdrop-api.md [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+
+# Examples
+
+## Enabling your application's window to be a drop target
+
+Let's say you are building the next greatest photo organizer app. To add photos to the library, you want to enable users to drag in files, for example from File Explorer to your app. To do this, you need to set your application's HWND/CoreWindow/AppWindow to be a drop target.
+
+1. Get `CoreDragDropManager`:
+
+This does a lot of initialization and basically informs the system that your app wants to register for drag & drop. Note that `CoreDragDropManager` is *per window*.
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: >  Note that `CoreDragDropManager` is *per window*.

Question: Does drag and drop only apply to top-level windows (windows without children)? Or if I have a window that's a [child of another window](https://docs.microsoft.com/en-us/windows/win32/winmsg/about-windows#parent-or-owner-window-handle) (say a text edit control within a text editor window), can I register it as a drag source or drop target using CoreDragDropManager?

If the answer depends on whether it's a USER32.dll window (HWND) or CoreWindow or AppWindow, it might make sense to make note of that here.

File: specs/dragdrop-api/dragdrop-api.md [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+
+1a. If you have **classic HWND-based** application:
+
+```c#
+// C#
+HWND hwnd; // HWND you got from CreateWindow or CreateWindowEx
+var manager = CoreDragDropManager.GetForWindow(hwnd);
+```
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: I suggest you remove this C# and HWND example as it's confusing. This is because apps written in C# or another .NET managed language are less likely to be using HWNDs directly. This is especially true for apps that use Windows Presentation Foundation as their GUI framework, since WPF uses USER32.dll windows (and thus HWNDs) as little as possible, generally only for top-level windows.

File: specs/dragdrop-api/dragdrop-api.md [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+# Description
+
+ Drag & drop is an intuitive feature that make for delightful app experiences for end users. When you use Windows' UI frameworks such as XAML/WinUI, you get their built-in support for drag & drop with minimal overhead. However, if you are not using those, implementing drag & drop can be challenging. This Project Reunion SDK API aims to alleviate that challenge by providing a feature-rich drag & drop experience in an easy-to-use set of APIs.
+
+The Reunion SDK drag & drop API is very similar to Windows' built-in "UWP drag & drop" API in the `Windows.ApplicationModel.DataTransfer.DragDrop.Core` Windows Runtime namespace. In fact, in this initial version (as of mid-2020), the Reunion SDK drag & drop API is almost the same as `Windows.ApplicationModel.DataTransfer.DragDrop.Core`, except for these differences:
+
+1. The namespace is now `Microsoft.Reunion.DragDrop`.
+2. The addition of `GetForHWND()` API to register any Windows app as a drop target.
+
+# Examples
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: In my opinion, a single language is enough for each example. Our internal guidelines for Windows Runtime API specs say to prefer C#, and in this case I think that's fine (apart from one exception below).

File: specs/dragdrop-api/dragdrop-api.md [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+
+# Description
+
+ Drag & drop is an intuitive feature that make for delightful app experiences for end users. When you use Windows' UI frameworks such as XAML/WinUI, you get their built-in support for drag & drop with minimal overhead. However, if you are not using those, implementing drag & drop can be challenging. This Project Reunion SDK API aims to alleviate that challenge by providing a feature-rich drag & drop experience in an easy-to-use set of APIs.
+
+The Reunion SDK drag & drop API is very similar to Windows' built-in "UWP drag & drop" API in the `Windows.ApplicationModel.DataTransfer.DragDrop.Core` Windows Runtime namespace. In fact, in this initial version (as of mid-2020), the Reunion SDK drag & drop API is almost the same as `Windows.ApplicationModel.DataTransfer.DragDrop.Core`, except for these differences:
+
+1. The namespace is now `Microsoft.Reunion.DragDrop`.
+2. The addition of `GetForHWND()` API to register any Windows app as a drop target.
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: Why is this necessary? CoreDragDropManager.GetForCurrentView() is supposed to get "the core drag and drop manager associated with the currently visible application window." If the "currently visible" window is a USER32.dll/HWND window, does the method not work? If so, we should consider fixing that problem in our copy of GetForCurrentView() instead of adding GetForHwnd().

File: specs/dragdrop-api/dragdrop-api.md [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+2. Implement `ICoreDropOperationTarget`:
+
+```c#
+// C#
+
+class DropTarget : ICoreDropOperationTarget
+{
+    public IAsyncOperation<DataPackageOperation> EnterAsync(CoreDragInfo dragInfo, CoreDragUIOverride dragUIOverride)
+    {
+        throw new NotImplementedException();
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop-api.md]: Add a "TODO" or "your code here" comment to each method for clarity.

File: specs/dragdrop-api/dragdrop-api.md (hunk 12 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: I was thinking to add pictures for this. What I meant was that the modern drag & drop API just has richer support for customizing the drag UI - customizing the image of the object being dragged, adding captions, etc.

I will add pictures to clarify.

File: specs/dragdrop-api/dragdrop-api.md (hunk 13 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: Thank you for bringing this up. This is a great question, and something for me to think about. This usage of the UWP API is without prior art; at the moment I cannot say how the specifics of the hierarchy of windows affect the usages of this API. I will investigate.

File: specs/dragdrop-api/dragdrop-api.md (hunk 14 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: Agree, I especially disliked the notion of putting HWND in C# code. With the removal of the GetForHWND API, this should look a lot better, and I will be able to remove this.

File: specs/dragdrop-api/dragdrop-api.md (hunk 16 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: Agree. Sudipto also pointed out the same thing. I will endeavor to remove `GetforHwnd` completely, and think about the naming of `GetForCurrentView`, possibly renaming to `GetForCurrentWindow` depending on the differences between HWND/CoreWindow/AppWindow.

File: specs/dragdrop-api/dragdrop.idl (hunk 8 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: To the best of my understanding, concurrent drag operations are currently only supported by CShell, which doesn't run on Desktop. Since Project Reunion is a primarily Desktop-targeted API, I will remove it from here for now.

File: specs/dragdrop-api/dragdrop-api.md (hunk 12 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: Added a better description for now, images are hard...

File: specs/dragdrop-api/dragdrop.idl [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+        DragUIContentMode DragUIContentMode;
+        void SetPointerId(UInt32 pointerId);
+        void SetDragUIContentFromSoftwareBitmap(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap);
+        void SetDragUIContentFromSoftwareBitmapWithAnchorPoint(Windows.Graphics.Imaging.SoftwareBitmap softwareBitmap, Windows.Foundation.Point anchorPoint);
+        Windows.Foundation.IAsyncOperation<Microsoft.ProjectReunion.ApplicationModel.DataPackageOperation> StartAsync();
+    }
+
+    runtimeclass DragDropManager
+    {
+        DragDropManager GetForCurrentWindow();
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >GetForCurrentWindow [](start = 24, length = 19)

this method should be static

File: specs/dragdrop-api/dragdrop.idl [hunk 19]
```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corp. Licensed under the MIT license.
+
+namespace Microsoft.ProjectReunion.ApplicationModel.DragDrop
```
Comment (usvoyager) [inline specs/dragdrop-api/dragdrop.idl]: >DragDrop [](start = 52, length = 8)

do you need this extra scope

Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: It's not necessary I think. Seems more of a design style question. The drag & drop members in this API are distinctly enough named that a separate namespace is not necessary. I don't have a preference either way

File: specs/dragdrop-api/dragdrop.idl [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+
+    runtimeclass DragDropManager
+    {
+        DragDropManager GetForCurrentWindow();
+        event Windows.Foundation.TypedEventHandler<DragDropManager, DropOperationTargetRequestedEventArgs> TargetRequested;
+    }
+
+    runtimeclass DropOperationTargetRequestedEventArgs
+    {
+        DropOperationTarget DropTarget { set; };
```
Comment (metathinker) [inline specs/dragdrop-api/dragdrop.idl]: I don't think I've ever seen a write-only WinRT property. Maybe make it read-write, or restore the SetDropTarget method it replaces?

Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: I can change it back to be a SetTarget method that takes a DropOperationTarget parameter. Out of curiosity, is it bad convention to have write-only property?

Comment (metathinker) [reply specs/dragdrop-api/dragdrop.idl]: I rarely see write-only properties myself in languages that have classes with properties as a concept; my personal expectation is that I can always read a property at least.

I don't have a strong opinion against write-only properties, though I can see how they can make code a bit more awkward:

```
DropOperationTargetRequestedEventArgs eventArgs;
// ...
eventArgs.DropTarget = new MyDropOperationTarget(1, 2, 3);
eventArgs.DropTarget.DoStuff(); // compile-time error???
```

Microsoft's C# and Visual Basic .NET compilers produce a static code analysis warning for write-only properties:
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1044?view=vs-2019
because "without read access, the state of shared objects cannot be viewed, which limits their usefulness." The above example I gave might be what was meant.

Comment (metathinker) [reply specs/dragdrop-api/dragdrop.idl]: Whether you bring back the method or leave the property, it should take an `IDropOperationTarget` instead of a `DropOperationTarget` since the API consumer is expected to implement `IDropOperationTarget` with their own code.

Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: correct, I noticed that about `IDropOperationTarget` too after I said it in the previous comment, thank you for pointing that out.

Thank you for the explanation; I see the possible confusion. I will bring back the method, as there is no reason the API consumer should need to inspect DropTarget as a property on the event args.

Comment (metathinker) [reply specs/dragdrop-api/dragdrop.idl]: > there is no reason the API consumer should need to inspect DropTarget as a property on the event args.

unless they set it themselves previously, as in my example. :) But that's easy to work around.

File: specs/dragdrop-api/dragdrop-api.md (hunk 13 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop-api.md]: Added a remark at the end to note that the API internally does the job of identifying what type of window it's being called with. This however does not answer your original question. I will keep this in mind and provide clarity as we go

File: specs/dragdrop-api/dragdrop.idl (hunk 19 above)
Comment (Ram-G) [reply specs/dragdrop-api/dragdrop.idl]: Removed this extra scope for now

File: specs/dragdrop-api/dragdrop.idl [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        static DragDropManager GetForCurrentWindow();
+        event Windows.Foundation.TypedEventHandler<DragDropManager, DropOperationTargetRequestedEventArgs> TargetRequested;
+    }
+
+    runtimeclass DropOperationTargetRequestedEventArgs
+    {
+        void SetTarget(IDropOperationTarget target);
+    }
+} // Microsoft.ProjectReunion.ApplicationModel.DragDrop
```
Comment (Ram-G) [inline specs/dragdrop-api/dragdrop.idl]: change namespace

**Comment by Ram-G** [comment]
Closing in favor of public review at https://github.com/microsoft/ProjectReunion/pull/141/.
